Sources/BazelizeKit/Codegen/Codegen+Application.swift

```diff
@@ -6,110 +6,137 @@
-            "minimum_os_version" => setting.macOS
-            "infoplists" => setting.infoPlist
-            "deps" => ":\(name)_library"
+            "bundle_id" => prefer(\.bundleID)
+            "minimum_os_version" => prefer(\.macOS)
+            "infoplists" => {
+                plist_file
+                plist_auto
+                // plist_default
+            }
+            "deps" => {
+                ":\(name)_library"
+            }
             StarlarkProperty.Visibility.public
         }
-
-        return builder.build()
     }
 
     /// ios_application(name, additional_linker_inputs, alternate_icons, app_clips, app_icons, bundle_id, bundle_name, codesign_inputs, codesignopts, deps, entitlements, entitlements_validation, executable_name, exported_symbols_lists, extensions, families, frameworks, include_symbols_in_bundle, infoplists, ipa_post_processor, launch_images, launch_storyboard, linkopts, minimum_deployment_os_version, minimum_os_version, platform_type, provisioning_profile, resources, sdk_frameworks, settings_bundle, stamp, strings, version, watch_application)
-    func buildIOS(_ builder: inout Build.Builder) {
+    func buildIOS(_ builder: inout Build.Builder, _: Kit) {
         builder.load(.ios_application)
         builder.add(.ios_application) {
             "name" => name
-            "bundle_id" => setting.bundleID
-            "families" => setting.deviceFamily
-            "minimum_os_version" => setting.iOS
-            "infoplists" => setting.infoPlist
+            "bundle_id" => prefer(\.bundleID)
+            "families" => prefer(\.deviceFamily)
+            "minimum_os_version" => prefer(\.iOS)
+            "infoplists" => {
+                plist_file
+                plist_auto
+                plist_default
+            }
             "launch_storyboard" => ":Base.lproj/LaunchScreen.storyboard"
-            "deps" => ":\(name)_library"
-            "frameworks" => frameworks
-            "sdk_frameworks" => sdkFrameworks
-            "resources" => resources
+            "deps" => {
+                ":\(name)_library"
+            }
+            "frameworks" => {
+                frameworks
+            }
+            "sdk_frameworks" => {
+                sdkFrameworks
+            }
+            "resources" => {
+                resources
+            }
             StarlarkProperty.Visibility.public
         }
     }
 
     /// macos_application(name, additional_contents, additional_linker_inputs, app_icons, bundle_extension, bundle_id, bundle_name, codesign_inputs, codesignopts, deps, entitlements, entitlements_validation, executable_name, exported_symbols_lists, extensions, include_symbols_in_bundle, infoplists, ipa_post_processor, linkopts, minimum_deployment_os_version, minimum_os_version, platform_type, provisioning_profile, resources, stamp, strings, version, xpc_services)
-    func buildMac(_ builder: inout Build.Builder) {
+    func buildMac(_ builder: inout Build.Builder, _: Kit) {
```

Comment:
`inout` used here is not perfect for Swift concurrency. Maybe in the future we need to make it more functional (replace `inout` with others)?

Reply:
Please can you provide the link that `inout` is not perfect for concurrency. 

Thx.

Reply:
In Current implementation, code generation was generated at same thread.

In this situation, is `inout` still influence `Swift concurrency`?

Reply:
- https://stackoverflow.com/questions/70748073/actor-isolated-property-cannot-be-passed-inout-to-async-function-call
- https://developer.apple.com/forums/thread/91694
`inout` is like a pointer here, which may block us from using concurrency features in the future.

Reply:
thx.